backend/main.py

```
@@ -71,151 +137,332 @@ def get_spots(
+        icon_url=user.icon_url,
+        wallet=schemas.UserWallet(coins=user.coins),
+        current_skin=schemas.SkinInfo(
+            id=current_skin.id,
+            name=current_skin.name,
+            image_url=current_skin.image_url
+        )
     )
 
+@app.post("/users/me/icon")
+async def update_user_icon(
+    file: UploadFile = File(...),
```

Ruff (0.14.7)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)